### PR TITLE
Recorder support for OS

### PR DESCRIPTION
**Pull request by manikanta-reddy-t**

### CODE REVIEW

File: server/src/main/java/com/testsigma/mapper/recorder/KibbutzPluginNLPMapper.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import com.testsigma.model.recorder.KibbutzPluginNLPParameterDTO;
+import com.testsigma.model.recorder.KibbutzPluginTestDataFunctionDTO;
+import org.mapstruct.*;
+
+@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS)
+public interface KibbutzPluginNLPMapper {
+/*
+    @Mapping(target = "applicationType", source = "workspaceType")
```
Comment (rajesh-testsigma) [inline server/src/main/java/com/testsigma/mapper/recorder/KibbutzPluginNLPMapper.java:17]: Remove commented out code

File: server/src/main/java/com/testsigma/model/TestStep.java [hunk 2]
```diff
@@ -234,6 +242,105 @@ public TestStepDataMap getDataMapBean() {
...
+      testStepForLoop.setTestDataId(forLoopTestDataId);
+      testStepForLoop.setStartIndex(forLoopStartIndex);
+      testStepForLoop.setEndIndex(forLoopEndIndex);
+      testStepDataMap.setForLoop(testStepForLoop);
+    }
+    return testStepDataMap;
+  }
+
+  /*
+  public TestStepRecorderDataMap getDataMap() {
```
Comment (rajesh-testsigma) [inline server/src/main/java/com/testsigma/model/TestStep.java:286]: Remove commented code

File: server/src/main/java/com/testsigma/model/recorder/ApplicationDTO.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.sql.Timestamp;
+
+@Data
+public class ApplicationDTO {
+    private Long id;
+    private String name;
+    private String description;
+    private String customFields;
+    //private Long projectId;
```
Comment (rajesh-testsigma) [inline server/src/main/java/com/testsigma/model/recorder/ApplicationDTO.java:14]: Remove project references commented code

File: server/src/main/java/com/testsigma/model/recorder/UiIdentifierRequest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    private ElementRequest[] siblings;
+    private ElementRequest[] parents;
+    private ElementScreenNameDTO screenNameObj;
+    private String currentImgBase64;
+    private String currentElementSource;
+    private Long shadowParentId;
+    private List<UiIdentifierRequest> shadowParentElements = new ArrayList<>();
+    private Long shadowOrder;
+
+    /*public boolean isDraft() {
```
Comment (rajesh-testsigma) [inline server/src/main/java/com/testsigma/model/recorder/UiIdentifierRequest.java:58]: renove commented code

**Review by rajesh-testsigma — COMMENTED** [review]
Looks good except the commented code

File: server/src/main/java/com/testsigma/mapper/recorder/KibbutzPluginNLPMapper.java (hunk 1 above)
Comment (tarun-testsigma) [reply server/src/main/java/com/testsigma/mapper/recorder/KibbutzPluginNLPMapper.java]: Removed comments

File: server/src/main/java/com/testsigma/model/TestStep.java (hunk 2 above)
Comment (tarun-testsigma) [reply server/src/main/java/com/testsigma/model/TestStep.java]: Removed comments

File: server/src/main/java/com/testsigma/model/recorder/ApplicationDTO.java (hunk 3 above)
Comment (tarun-testsigma) [reply server/src/main/java/com/testsigma/model/recorder/ApplicationDTO.java]: Removed comments

File: server/src/main/java/com/testsigma/model/recorder/UiIdentifierRequest.java (hunk 4 above)
Comment (tarun-testsigma) [reply server/src/main/java/com/testsigma/model/recorder/UiIdentifierRequest.java]: Removed comments

**Review by rajesh-testsigma — APPROVED** [review]

**Pull request merged**